src/service/alert.service.ts

```diff
@@ -29,7 +29,7 @@ export class AlertService {
     }
 
     clear() {
-        this.alerts = [];
+        this.alerts.forEach((alert: Alert) => setTimeout(() => alert.close(), 0));
```

Comment:
Yes. The alerts array sets a new empty array. But the alert.component has a reference to the original array alerts.